// ionic-context-menu/esm5/lib/ionic-context-menu.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, Input } from '@angular/core';
import { PopoverController } from '@ionic/angular';
import { IonicContextMenuContentComponent } from './ionic-context-menu-content.component';
var IonicContextMenuComponent = /** @class */ (function () {
    function IonicContextMenuComponent(popoverCtrl) {
        this.popoverCtrl = popoverCtrl;
    }
    /**
     * @param {?} child
     * @return {?}
     */
    IonicContextMenuComponent.prototype.present = /**
     * @param {?} child
     * @return {?}
     */
    function (child) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var _a;
            return tslib_1.__generator(this, function (_b) {
                switch (_b.label) {
                    case 0:
                        _a = this;
                        return [4 /*yield*/, this.popoverCtrl.create({
                                component: IonicContextMenuContentComponent,
                                componentProps: {
                                    template: child
                                },
                                event: event,
                                translucent: true
                            })];
                    case 1:
                        _a.popover = _b.sent();
                        return [4 /*yield*/, this.popover.present()];
                    case 2: return [2 /*return*/, _b.sent()];
                }
            });
        });
    };
    IonicContextMenuComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ionic-context-menu',
                    template: "\n    <div>\n      <ion-button (click)=\"present(template)\">\n        <ion-icon slot=\"end\" [name]=\"icon || 'more'\"></ion-icon>\n      </ion-button>\n      <ng-template #template>\n        <div (click)=\"popover.dismiss()\">\n          <ng-content></ng-content>\n        </div>\n      </ng-template>\n    </div>\n  ",
                    styles: []
                },] },
    ];
    /** @nocollapse */
    IonicContextMenuComponent.ctorParameters = function () { return [
        { type: PopoverController }
    ]; };
    IonicContextMenuComponent.propDecorators = {
        icon: [{ type: Input }]
    };
    return IonicContextMenuComponent;
}());
export { IonicContextMenuComponent };
if (false) {
    /** @type {?} */
    IonicContextMenuComponent.prototype.popover;
    /** @type {?} */
    IonicContextMenuComponent.prototype.icon;
    /** @type {?} */
    IonicContextMenuComponent.prototype.popoverCtrl;
}
