// ionic-context-menu/fesm2015/ionic-context-menu.js
import { Component, Input, NgModule } from '@angular/core';
import { __awaiter } from 'tslib';
import { PopoverController, IonicModule } from '@ionic/angular';
import { CommonModule } from '@angular/common';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
class IonicContextMenuContentComponent {
}
IonicContextMenuContentComponent.decorators = [
    { type: Component, args: [{
                selector: 'ionic-context-menu-content',
                template: `
  <ng-template #defaultTemplate>
    <div></div>
  </ng-template>
  <ng-container *ngTemplateOutlet="template ? template : defaultTemplate"></ng-container>
`
            },] },
];
IonicContextMenuContentComponent.propDecorators = {
    template: [{ type: Input }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
class IonicContextMenuComponent {
    /**
     * @param {?} popoverCtrl
     */
    constructor(popoverCtrl) {
        this.popoverCtrl = popoverCtrl;
    }
    /**
     * @param {?} child
     * @return {?}
     */
    present(child) {
        return __awaiter(this, void 0, void 0, function* () {
            this.popover = yield this.popoverCtrl.create({
                component: IonicContextMenuContentComponent,
                componentProps: {
                    template: child
                },
                event,
                translucent: true
            });
            return yield this.popover.present();
        });
    }
}
IonicContextMenuComponent.decorators = [
    { type: Component, args: [{
                selector: 'ionic-context-menu',
                template: `
    <div>
      <ion-button (click)="present(template)">
        <ion-icon slot="end" [name]="icon || 'more'"></ion-icon>
      </ion-button>
      <ng-template #template>
        <div (click)="popover.dismiss()">
          <ng-content></ng-content>
        </div>
      </ng-template>
    </div>
  `,
                styles: []
            },] },
];
/** @nocollapse */
IonicContextMenuComponent.ctorParameters = () => [
    { type: PopoverController }
];
IonicContextMenuComponent.propDecorators = {
    icon: [{ type: Input }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
class IonicContextMenuItemComponent {
}
IonicContextMenuItemComponent.decorators = [
    { type: Component, args: [{
                selector: 'ionic-context-menu-item',
                template: `
  <ion-item lines="none">
    <ion-icon [name]="icon" slot="start"></ion-icon>
    <ion-label>
      <ng-content></ng-content>
    </ion-label>
  </ion-item>
`
            },] },
];
IonicContextMenuItemComponent.propDecorators = {
    icon: [{ type: Input }]
};

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
class IonicContextMenuModule {
}
IonicContextMenuModule.decorators = [
    { type: NgModule, args: [{
                imports: [IonicModule, CommonModule],
                declarations: [IonicContextMenuComponent, IonicContextMenuContentComponent, IonicContextMenuItemComponent],
                exports: [IonicContextMenuComponent, IonicContextMenuContentComponent, IonicContextMenuItemComponent],
                entryComponents: [IonicContextMenuComponent, IonicContextMenuContentComponent, IonicContextMenuItemComponent]
            },] },
];

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

export { IonicContextMenuComponent, IonicContextMenuModule, IonicContextMenuContentComponent as ɵa, IonicContextMenuItemComponent as ɵb };
