// ionic-context-menu/fesm5/ionic-context-menu.js
import { Component, Input, NgModule } from '@angular/core';
import { __awaiter, __generator } from 'tslib';
import { PopoverController, IonicModule } from '@ionic/angular';
import { CommonModule } from '@angular/common';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
var IonicContextMenuContentComponent = /** @class */ (function () {
    function IonicContextMenuContentComponent() {
    }
    IonicContextMenuContentComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ionic-context-menu-content',
                    template: "\n  <ng-template #defaultTemplate>\n    <div></div>\n  </ng-template>\n  <ng-container *ngTemplateOutlet=\"template ? template : defaultTemplate\"></ng-container>\n"
                },] },
    ];
    IonicContextMenuContentComponent.propDecorators = {
        template: [{ type: Input }]
    };
    return IonicContextMenuContentComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
var IonicContextMenuComponent = /** @class */ (function () {
    function IonicContextMenuComponent(popoverCtrl) {
        this.popoverCtrl = popoverCtrl;
    }
    /**
     * @param {?} child
     * @return {?}
     */
    IonicContextMenuComponent.prototype.present = /**
     * @param {?} child
     * @return {?}
     */
    function (child) {
        return __awaiter(this, void 0, void 0, function () {
            var _a;
            return __generator(this, function (_b) {
                switch (_b.label) {
                    case 0:
                        _a = this;
                        return [4 /*yield*/, this.popoverCtrl.create({
                                component: IonicContextMenuContentComponent,
                                componentProps: {
                                    template: child
                                },
                                event: event,
                                translucent: true
                            })];
                    case 1:
                        _a.popover = _b.sent();
                        return [4 /*yield*/, this.popover.present()];
                    case 2: return [2 /*return*/, _b.sent()];
                }
            });
        });
    };
    IonicContextMenuComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ionic-context-menu',
                    template: "\n    <div>\n      <ion-button (click)=\"present(template)\">\n        <ion-icon slot=\"end\" [name]=\"icon || 'more'\"></ion-icon>\n      </ion-button>\n      <ng-template #template>\n        <div (click)=\"popover.dismiss()\">\n          <ng-content></ng-content>\n        </div>\n      </ng-template>\n    </div>\n  ",
                    styles: []
                },] },
    ];
    /** @nocollapse */
    IonicContextMenuComponent.ctorParameters = function () { return [
        { type: PopoverController }
    ]; };
    IonicContextMenuComponent.propDecorators = {
        icon: [{ type: Input }]
    };
    return IonicContextMenuComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
var IonicContextMenuItemComponent = /** @class */ (function () {
    function IonicContextMenuItemComponent() {
    }
    IonicContextMenuItemComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ionic-context-menu-item',
                    template: "\n  <ion-item lines=\"none\">\n    <ion-icon [name]=\"icon\" slot=\"start\"></ion-icon>\n    <ion-label>\n      <ng-content></ng-content>\n    </ion-label>\n  </ion-item>\n"
                },] },
    ];
    IonicContextMenuItemComponent.propDecorators = {
        icon: [{ type: Input }]
    };
    return IonicContextMenuItemComponent;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
var IonicContextMenuModule = /** @class */ (function () {
    function IonicContextMenuModule() {
    }
    IonicContextMenuModule.decorators = [
        { type: NgModule, args: [{
                    imports: [IonicModule, CommonModule],
                    declarations: [IonicContextMenuComponent, IonicContextMenuContentComponent, IonicContextMenuItemComponent],
                    exports: [IonicContextMenuComponent, IonicContextMenuContentComponent, IonicContextMenuItemComponent],
                    entryComponents: [IonicContextMenuComponent, IonicContextMenuContentComponent, IonicContextMenuItemComponent]
                },] },
    ];
    return IonicContextMenuModule;
}());

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */

export { IonicContextMenuComponent, IonicContextMenuModule, IonicContextMenuContentComponent as ɵa, IonicContextMenuItemComponent as ɵb };
