// ionic-context-menu/bundles/ionic-context-menu.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@ionic/angular'), require('@angular/common')) :
    typeof define === 'function' && define.amd ? define('ionic-context-menu', ['exports', '@angular/core', '@ionic/angular', '@angular/common'], factory) :
    (factory((global['ionic-context-menu'] = {}),global.ng.core,null,global.ng.common));
}(this, (function (exports,core,angular,common) { 'use strict';

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation. All rights reserved.
    Licensed under the Apache License, Version 2.0 (the "License"); you may not use
    this file except in compliance with the License. You may obtain a copy of the
    License at http://www.apache.org/licenses/LICENSE-2.0

    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
    WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
    MERCHANTABLITY OR NON-INFRINGEMENT.

    See the Apache Version 2.0 License for specific language governing permissions
    and limitations under the License.
    ***************************************************************************** */
    function __awaiter(thisArg, _arguments, P, generator) {
        return new (P || (P = Promise))(function (resolve, reject) {
            function fulfilled(value) { try {
                step(generator.next(value));
            }
            catch (e) {
                reject(e);
            } }
            function rejected(value) { try {
                step(generator["throw"](value));
            }
            catch (e) {
                reject(e);
            } }
            function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
            step((generator = generator.apply(thisArg, _arguments || [])).next());
        });
    }
    function __generator(thisArg, body) {
        var _ = { label: 0, sent: function () { if (t[0] & 1)
                throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
        return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function () { return this; }), g;
        function verb(n) { return function (v) { return step([n, v]); }; }
        function step(op) {
            if (f)
                throw new TypeError("Generator is already executing.");
            while (_)
                try {
                    if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done)
                        return t;
                    if (y = 0, t)
                        op = [op[0] & 2, t.value];
                    switch (op[0]) {
                        case 0:
                        case 1:
                            t = op;
                            break;
                        case 4:
                            _.label++;
                            return { value: op[1], done: false };
                        case 5:
                            _.label++;
                            y = op[1];
                            op = [0];
                            continue;
                        case 7:
                            op = _.ops.pop();
                            _.trys.pop();
                            continue;
                        default:
                            if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) {
                                _ = 0;
                                continue;
                            }
                            if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) {
                                _.label = op[1];
                                break;
                            }
                            if (op[0] === 6 && _.label < t[1]) {
                                _.label = t[1];
                                t = op;
                                break;
                            }
                            if (t && _.label < t[2]) {
                                _.label = t[2];
                                _.ops.push(op);
                                break;
                            }
                            if (t[2])
                                _.ops.pop();
                            _.trys.pop();
                            continue;
                    }
                    op = body.call(thisArg, _);
                }
                catch (e) {
                    op = [6, e];
                    y = 0;
                }
                finally {
                    f = t = 0;
                }
            if (op[0] & 5)
                throw op[1];
            return { value: op[0] ? op[1] : void 0, done: true };
        }
    }

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var IonicContextMenuContentComponent = /** @class */ (function () {
        function IonicContextMenuContentComponent() {
        }
        IonicContextMenuContentComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ionic-context-menu-content',
                        template: "\n  <ng-template #defaultTemplate>\n    <div></div>\n  </ng-template>\n  <ng-container *ngTemplateOutlet=\"template ? template : defaultTemplate\"></ng-container>\n"
                    },] },
        ];
        IonicContextMenuContentComponent.propDecorators = {
            template: [{ type: core.Input }]
        };
        return IonicContextMenuContentComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var IonicContextMenuComponent = /** @class */ (function () {
        function IonicContextMenuComponent(popoverCtrl) {
            this.popoverCtrl = popoverCtrl;
        }
        /**
         * @param {?} child
         * @return {?}
         */
        IonicContextMenuComponent.prototype.present = /**
         * @param {?} child
         * @return {?}
         */
            function (child) {
                return __awaiter(this, void 0, void 0, function () {
                    var _a;
                    return __generator(this, function (_b) {
                        switch (_b.label) {
                            case 0:
                                _a = this;
                                return [4 /*yield*/, this.popoverCtrl.create({
                                        component: IonicContextMenuContentComponent,
                                        componentProps: {
                                            template: child
                                        },
                                        event: event,
                                        translucent: true
                                    })];
                            case 1:
                                _a.popover = _b.sent();
                                return [4 /*yield*/, this.popover.present()];
                            case 2: return [2 /*return*/, _b.sent()];
                        }
                    });
                });
            };
        IonicContextMenuComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ionic-context-menu',
                        template: "\n    <div>\n      <ion-button (click)=\"present(template)\">\n        <ion-icon slot=\"end\" [name]=\"icon || 'more'\"></ion-icon>\n      </ion-button>\n      <ng-template #template>\n        <div (click)=\"popover.dismiss()\">\n          <ng-content></ng-content>\n        </div>\n      </ng-template>\n    </div>\n  ",
                        styles: []
                    },] },
        ];
        /** @nocollapse */
        IonicContextMenuComponent.ctorParameters = function () {
            return [
                { type: angular.PopoverController }
            ];
        };
        IonicContextMenuComponent.propDecorators = {
            icon: [{ type: core.Input }]
        };
        return IonicContextMenuComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var IonicContextMenuItemComponent = /** @class */ (function () {
        function IonicContextMenuItemComponent() {
        }
        IonicContextMenuItemComponent.decorators = [
            { type: core.Component, args: [{
                        selector: 'ionic-context-menu-item',
                        template: "\n  <ion-item lines=\"none\">\n    <ion-icon [name]=\"icon\" slot=\"start\"></ion-icon>\n    <ion-label>\n      <ng-content></ng-content>\n    </ion-label>\n  </ion-item>\n"
                    },] },
        ];
        IonicContextMenuItemComponent.propDecorators = {
            icon: [{ type: core.Input }]
        };
        return IonicContextMenuItemComponent;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */
    var IonicContextMenuModule = /** @class */ (function () {
        function IonicContextMenuModule() {
        }
        IonicContextMenuModule.decorators = [
            { type: core.NgModule, args: [{
                        imports: [angular.IonicModule, common.CommonModule],
                        declarations: [IonicContextMenuComponent, IonicContextMenuContentComponent, IonicContextMenuItemComponent],
                        exports: [IonicContextMenuComponent, IonicContextMenuContentComponent, IonicContextMenuItemComponent],
                        entryComponents: [IonicContextMenuComponent, IonicContextMenuContentComponent, IonicContextMenuItemComponent]
                    },] },
        ];
        return IonicContextMenuModule;
    }());

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */

    /**
     * @fileoverview added by tsickle
     * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
     */

    exports.IonicContextMenuComponent = IonicContextMenuComponent;
    exports.IonicContextMenuModule = IonicContextMenuModule;
    exports.ɵa = IonicContextMenuContentComponent;
    exports.ɵb = IonicContextMenuItemComponent;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
