// ionic-context-menu/esm2015/lib/ionic-context-menu.component.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, Input } from '@angular/core';
import { PopoverController } from '@ionic/angular';
import { IonicContextMenuContentComponent } from './ionic-context-menu-content.component';
export class IonicContextMenuComponent {
    /**
     * @param {?} popoverCtrl
     */
    constructor(popoverCtrl) {
        this.popoverCtrl = popoverCtrl;
    }
    /**
     * @param {?} child
     * @return {?}
     */
    present(child) {
        return tslib_1.__awaiter(this, void 0, void 0, function* () {
            this.popover = yield this.popoverCtrl.create({
                component: IonicContextMenuContentComponent,
                componentProps: {
                    template: child
                },
                event,
                translucent: true
            });
            return yield this.popover.present();
        });
    }
}
IonicContextMenuComponent.decorators = [
    { type: Component, args: [{
                selector: 'ionic-context-menu',
                template: `
    <div>
      <ion-button (click)="present(template)">
        <ion-icon slot="end" [name]="icon || 'more'"></ion-icon>
      </ion-button>
      <ng-template #template>
        <div (click)="popover.dismiss()">
          <ng-content></ng-content>
        </div>
      </ng-template>
    </div>
  `,
                styles: []
            },] },
];
/** @nocollapse */
IonicContextMenuComponent.ctorParameters = () => [
    { type: PopoverController }
];
IonicContextMenuComponent.propDecorators = {
    icon: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    IonicContextMenuComponent.prototype.popover;
    /** @type {?} */
    IonicContextMenuComponent.prototype.icon;
    /** @type {?} */
    IonicContextMenuComponent.prototype.popoverCtrl;
}
